Makefile

```diff
@@ -73,7 +74,9 @@ $(EXAMPLE_DIRS): lib
 examples: $(EXAMPLE_DIRS)
 	$(Q)true
 
-clean: $(EXAMPLE_DIRS:=.clean) styleclean
+examplesclean: $(EXAMPLE_DIRS:=.clean) styleclean
+
+clean: examplesclean
 	$(Q)$(MAKE) -C libopencm3 clean
```

Comment:
Do you want this cascade? or are there multiple clean targets now: clean (all) examplesclean, styleclean, libclean?, docclean?